pkg/generators/golang/types_generator.go

```diff
@@ -313,13 +316,20 @@ func (g *TypesGenerator) generateStructTypeSource(typ *concepts.Type) {
 		//
 		{{ lineComment .Type.Doc }}
 		type  {{ $objectName }} struct {
+			bitmap_ {{ bitmapType .Type }}
 			{{ if .Type.IsClass }}
-				id   *string
-				href *string
-				link bool
+				id   string
+				href string
 			{{ end }}
 			{{ range .Type.Attributes }}
-				{{ fieldName . }} {{ fieldType . }}
+				{{ if not .Type.IsBoolean }}
+					{{ fieldName . }} {{ fieldType . }}
+				{{ end }}
+			{{ end }}
+			{{ range .Type.Attributes }}
+				{{ if .Type.IsBoolean }}
+					{{ fieldName . }} {{ fieldType . }}
+				{{ end }}
```

Comment:
This looks like exactly same code duplicated twice?

Reply:
Ah, I missed the `if not` vs `if`.

I suppose this is to pack the struct better due to alignment constraints? :+1:

Reply:
LGTM